cfme/infrastructure/pxe.py

```diff
@@ -522,48 +777,138 @@ def delete(self, cancel=True):
-        sel.handle_alert(cancel=cancel)
-        flash.assert_message_match('System Image Type "{}": Delete successful'.format(self.name))
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove this System Image Type',
+                                               handle_alert=not cancel)
+        if not cancel:
+            main_view = self.create_view(PXESystemImageTypesView)
+            msg = 'System Image Type "{}": Delete successful'.format(self.name)
+            main_view.flash.assert_success_message(msg)
 
 
 @navigator.register(SystemImageType, 'All')
 class SystemImageTypeAll(CFMENavigateStep):
+    VIEW = PXESystemImageTypesView
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
     def step(self):
         self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'PXE')
-        acc.tree("System Image Types", "All System Image Types")
+        self.view.sidebar.image_types.tree.click_path('All System Image Types')
 
 
 @navigator.register(SystemImageType, 'Add')
 class SystemImageTypeAdd(CFMENavigateStep):
+    VIEW = PXESystemImageTypeAddView
     prerequisite = NavigateToSibling('All')
 
     def step(self):
-        cfg_btn('Add a new System Image Type')
+        self.prerequisite_view.toolbar.configuration.item_select('Add a new System Image Type')
 
 
 @navigator.register(SystemImageType, 'Details')
 class SystemImageTypeDetails(CFMENavigateStep):
+    VIEW = PXESystemImageTypeDetailsView
     prerequisite = NavigateToSibling('All')
 
     def step(self):
-        image_table.click_cell('name', self.obj.name)
+        self.prerequisite_view.sidebar.image_types.tree.click_path('All System Image Types',
+                                                                   self.obj.name)
 
 
 @navigator.register(SystemImageType, 'Edit')
 class SystemImageTypeEdit(CFMENavigateStep):
+    VIEW = PXESystemImageTypeEditView
     prerequisite = NavigateToSibling('Details')
 
     def step(self):
-        cfg_btn('Edit this System Image Type')
+        self.prerequisite_view.toolbar.configuration.item_select('Edit this System Image Type')
+
+
+class PXEDatastoreEntity(ParametrizedView, ClickableMixin):
+    """
+     represents ISO Datastore Entity
+    """
+    PARAMETERS = ('name', )
+    ROOT = ParametrizedLocator('.//tr[./td[contains(normalize-space(.), {name|quote})]]')
+
+    checkbox = Checkbox(locator='./td/input[@type="checkbox"]')
+    name = Text(locator=ParametrizedString('./td[contains(normalize-space(.), {name|quote})]'))
+    provision_type = Text(locator=ParametrizedString('./td[contains(normalize-space(.), '
+                                                     '{name|quote})]/following-sibling::td'))
+
+    def check(self):
+        self.checkbox.fill(True)
+
+    def uncheck(self):
+        self.checkbox.fill(False)
+
+    @property
+    def is_checked(self):
+        return self.checkbox.selected
```

Comment:
Same here with the Table, can't the Table from widgetastic_manageiq be used here?